src/sentry/incidents/endpoints/serializers.py

```diff
@@ -151,189 +353,117 @@ def update(self, instance, validated_data):
-     - `organization`: The organization related to this trigger.
+    Unified Serializer for creating/updating an alert rule - accepts trigger and action data, and does validation on it.
+    Required context:
+     - `organization`: The organization related to this alert rule.
      - `access`: An access object (from `request.access`)
     """
 
-    # TODO: These might be slow for many projects, since it will query for each
-    # individually. If we find this to be a problem then we can look into batching.
-    excluded_projects = serializers.ListField(child=ProjectField(), required=False)
-
-    class Meta:
-        model = AlertRuleTrigger
-        fields = [
-            "label",
-            "threshold_type",
-            "alert_threshold",
-            "resolve_threshold",
-            "excluded_projects",
-        ]
-        extra_kwargs = {"label": {"min_length": 1, "max_length": 64}}
-
-    def validate_threshold_type(self, threshold_type):
-        try:
-            return AlertRuleThresholdType(threshold_type)
-        except ValueError:
-            raise serializers.ValidationError(
-                "Invalid threshold type, valid values are %s"
-                % [item.value for item in AlertRuleThresholdType]
-            )
+    triggers = AlertRuleTriggerSerializer(many=True, required=True)
+
+    class Meta(AlertRuleSerializer.Meta):
+        fields = AlertRuleSerializer.Meta.fields + ["triggers"]
+
+    def validate(self, data):
+        """Performs validation on an alert rule's data
+        This includes ensuring there is either 1 or 2 triggers, which each have actions, and have proper thresholds set.
+        The critical trigger should both alert and resolve 'after' the warning trigger (whether that means > or < the value depends on threshold type).
+        """
+        triggers = data.get('triggers', [])
+        if triggers:
+            if len(triggers) == 1:
+                if triggers[0].get('label', None) != CRITICAL_TRIGGER_LABEL:
+                    raise serializers.ValidationError('First trigger must be labeled "%s"' % (CRITICAL_TRIGGER_LABEL))
+            elif len(triggers) == 2:
+                critical = triggers[0]
+                warning = triggers[1]
+                if critical.get('label', None) != CRITICAL_TRIGGER_LABEL or warning['label'] != WARNING_TRIGGER_LABEL:
+                    raise serializers.ValidationError('First trigger must be labeled "%s", second trigger must be labeled "%s"' % (CRITICAL_TRIGGER_LABEL, WARNING_TRIGGER_LABEL))
+                else:
+                    if critical['threshold_type'] != warning['threshold_type']:
+                        raise serializers.ValidationError('Must have matching threshold types (i.e. critical and warning triggers must both be an upper or lower bound)')
+
+                    if critical['threshold_type'] == AlertRuleThresholdType.ABOVE:
+                        alert_op, resolve_op = operator.lt, operator.lt
```

Comment:
Ahh, didn't realize these were the same, we can just have op be a single variable then.

Reply:
They weren't originally, but the logic was incorrect (eep). Fixed the logic.